packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,432 @@
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'text.dart';
+import 'widget_span.dart';
+
+// TODO(justinmc): Add SelectableArea to the examples.
+
+// TODO(justinmc): On some platforms, may want to underline link?
+
+/// A callback that passes a [String] representing a URL.
+typedef UriStringCallback = void Function(String urlString);
+
+/// Builds an InlineSpan for displaying [linkText].
+///
+/// If tappable, then uses the [onTap] handler.
+///
+/// Typically used for styling a link in some inline text.
+typedef LinkBuilder = InlineSpan Function(
+  String linkText,
+  // TODO(justinmc): Rename to LinkCallback?
+  //UriStringCallback? onTap
+);
+
+// TODO(justinmc): Change name to something link-agnostic?
+// TODO(justinmc): Add regexp parameter. No, builder? Both?
+/// A widget that displays some text with parts of it made interactive.
+///
+/// By default, any URLs in the text are made interactive.
+///
+/// See also:
+///
+///  * [InlineLinkedText], which is like this but is an inline TextSpan instead
+///    of a widget.
+class LinkedText extends StatelessWidget {
+  /// Creates an instance of [LinkedText] from the given [text], highlighting
+  /// any URLs by default.
+  ///
+  /// {@template flutter.widgets.LinkedText.new}
+  /// By default, highlights URLs in the [text] and makes them tappable with
+  /// [onTap].
+  ///
+  /// If [ranges] is given, then makes those ranges in the text interactive
+  /// instead of URLs.
+  ///
+  /// [linkBuilder] can be used to specify a custom [InlineSpan] for each
+  /// [TextRange] in [ranges].
+  /// {@endtemplate}
+  ///
+  /// See also:
+  ///
+  ///  * [LinkedText.regExp], which automatically finds ranges that match
+  ///    the given [RegExp].
+  ///  * [LinkedText.textLinkers], which uses [TextLinker]s to allow
+  ///    specifying an arbitrary number of [ranges] and [linkBuilders].
+  ///  * [InlineLinkedText.new], which is like this, but for inline text.
+  LinkedText({
```

Comment:
Can this widget be replaced by text.rich if we provide the InlineLinkedText.textLinkers to convert inlinespan to inlinespan?

Reply:
Yes, you can just put an InlineLinkedText inside of a Text.rich and it will work.  See the example inline_linked_text.0.dart where I show how to do this.  I expect people that want more customization or are working with other InlineSpans will do this.

Probably the biggest benefit of using LinkedText is that it will automatically manage the GestureRecognizer lifecycle for you.  Otherwise, you have to `dispose` the recognizer created for each link by the InlineLinkedText, similar to [TextSpan.recognizer](https://master-api.flutter.dev/flutter/painting/TextSpan/recognizer.html).

Reply:
That's a good point that I should have thought about!  I was just playing around with both of these options and then thought of another: adding a `builder` parameter (https://github.com/flutter/flutter/pull/125927/commits/71cfaf979e68655b347829a590ed62ecb2fed432).  I'm hoping this is the best of both worlds.  Users can customize the Text.rich as much as they want, but I'm not polluting `Text` with constructors or `LinkedText` with parameters.  Let me know what you think.